### ISSUE TITLE
Icinga2 2.7.0 requires SELinux boolean icinga2_can_connect_all on CentOS 7 even for default port

### ISSUE THREAD

**Issue by craigafinch** [Icinga/icinga2#5479]
I ran into the same [SELinux issue that was reported in this thread](https://monitoring-portal.org/index.php?thread/41507-icinga-2-7-0-selinux-networking-issues/). Since no issue has been opened yet, I'm opening one now. Information below comes from the original post; my issue is identical.

## Expected Behavior

Icinga2 on central monitor server connects to Icinga2 clients to monitor attributes of client servers

## Current Behavior

On CentOS 7, upgrading Icinga2 to 2.7.0 results in SELinux denials that prevent the master server  from contacting icinga2 on other hosts:

Entries in `/var/log/icinga2/icinga2.log`:

```
[2017-08-08 20:23:25 +0000] critical/TcpSocket: Invalid socket: Connection refused
[2017-08-08 20:23:25 +0000] critical/ApiListener: Cannot connect to host 'xxx' on port '5665'
[2017-08-08 20:24:25 +0000] information/ApiListener: Reconnecting to API endpoint 'xxx' via host 'xxx' and port '5665'
```

```
grep name_connect /var/log/audit/audit.log |tail -n1 |audit2why
type=AVC msg=audit(1502204405.955:5660): avc:  denied  { name_connect } for  pid=2488 comm="icinga2" dest=5665 scontext=system_u:system_r:icinga2_t:s0 tcontext=system_u:object_r:icinga2_port_t:s0 tclass=tcp_socket

        Was caused by:
        The boolean icinga2_can_connect_all was set incorrectly.
        Description:
        Allow icinga2 to can connect all

        Allow access by executing:
        # setsebool -P icinga2_can_connect_all 1
```

## Possible Solution (Workaround)
Set the suggested SELinux boolean

## Steps to Reproduce (for bugs)
1. Run a fully up-to-date CentOS 7 server (7.3.1611)
2. Update Icinga2 to 2.7.0
3. Attempt to connect to Icinga2 on a client machine

## Your Environment

* Version used (`icinga2 --version`):
```
icinga2 --version
icinga2 - The Icinga 2 network monitoring daemon (version: r2.7.0-1)

Copyright (c) 2012-2017 Icinga Development Team (https://www.icinga.com/)
License GPLv2+: GNU GPL version 2 or later <http://gnu.org/licenses/gpl2.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Application information:
  Installation root: /usr
  Sysconf directory: /etc
  Run directory: /run
  Local state directory: /var
  Package data directory: /usr/share/icinga2
  State path: /var/lib/icinga2/icinga2.state
  Modified attributes path: /var/lib/icinga2/modified-attributes.conf
  Objects path: /var/cache/icinga2/icinga2.debug
  Vars path: /var/cache/icinga2/icinga2.vars
  PID path: /run/icinga2/icinga2.pid

System information:
  Platform: CentOS Linux
  Platform version: 7 (Core)
  Kernel: Linux
  Kernel version: 3.10.0-514.26.2.el7.x86_64
  Architecture: x86_64

Build information:
  Compiler: GNU 4.8.5
  Build host: unknown
```
* Operating System and version:
```
cat /etc/redhat-release
CentOS Linux release 7.3.1611 (Core)

Installed Packages
Name        : icinga2-selinux
Arch        : x86_64
Version     : 2.7.0
Release     : 1.el7.icinga
Size        : 2.7 M
Repo        : installed
From repo   : icinga-stable-release
Summary     : SELinux policy module supporting icinga2
URL         : https://www.icinga.com/
License     : GPL-2.0+
Description : SELinux policy module supporting icinga2
```
* Enabled features (`icinga2 feature list`):
```
icinga2 feature list
Disabled features: compatlog debuglog gelf graphite influxdb livestatus opentsdb perfdata statusdata syslog
Enabled features: api checker command ido-mysql mainlog notification
```

**Comment by dgoetz** [comment]
I will have a look into as setting this boolean should not be required as the policy includes rules for opening the socket on the port.

```
allow icinga2_t icinga2_port_t:tcp_socket name_bind;
allow icinga2_t self:tcp_socket create_stream_socket_perms;
```

But this does not include `name_connect` as this is not opening the socket, it is connecting to an already opened socket.
@dnsmichi @gunnarbeutner: Has something changed the behaviour of the API feature in this way?

**Comment by dnsmichi** [comment]
@dgoetz not that I know of. But it might require a comparison test between 2.6.3 and 2.7.0 in this regard.

I've pointed the user from the forums to here, since you're the SELinux expert (and I am not).

**Comment by dgoetz** [comment]
I still not know why Icinga 2 now requires the permission but I could reproduce the problem. Perhaps it is something like the first process opening the socket and the second one using it. Policy now simply allows icinga2_t to do so.

While investigating I also found the new notification command requiring more privileges, now it is also allowed to execute `hostname`.

**Comment by dnsmichi** [comment]
Fixed in #5501 - please test the snapshot packages then.
